libr/core/cmd_debug.c

```diff
@@ -935,111 +936,158 @@ static void get_hash_debug_file(const char *path, char *hash, int hash_len) {
-	ut64 libc_addr = UT64_MAX;
-
-	if (!core || !core->dbg || !core->dbg->maps) return;
-	r_debug_map_sync (core->dbg);
-	r_list_foreach (core->dbg->maps, iter, map) {
-		if (strstr (map->name, "/libc-")) {
-			libc_addr = (SIZE_SZ == 4) ? (map->addr_end) : (map->addr);
-			libc_ver_end = map->name;
-			break;
+	RDebugMap *map;
+	
+	if (*m_arena == UT64_MAX) {
+		const char *dir_dbg = "/usr/lib/debug";
+		const char *dir_build_id = "/.build-id";
+		const char *symname = "main_arena";
+		const char *libc_ver_end = NULL; 
+		char hash[64] = {0}, path[1024] = {0};
+		char *custom_libc = NULL;
+		bool is_debug_file[2];
+		ut64 libc_addr = UT64_MAX;
+
+		if (!core || !core->dbg || !core->dbg->maps){
+			return 1;
 		}
-	}
-
-	if (!libc_ver_end) {
-		eprintf ("Warning: Is glibc mapped in memory? (see dm command)\n");
-		return;
-	}
-
-	is_debug_file[0] = str_start_with (libc_ver_end, "/usr/lib/");
-	is_debug_file[1] = str_start_with (libc_ver_end, "/lib/");
+		r_debug_map_sync (core->dbg);
+		r_list_foreach (core->dbg->maps, iter, map) {
+			if (strstr (map->name, "/libc-")) {
+				libc_addr = map->addr;
+				libc_ver_end = map->name;
+				break;
+			}
+		}
+		if (!libc_ver_end) {
+			eprintf ("Warning: Is glibc mapped in memory? (see dm command)\n");
+			return 1;
+		}
+		is_debug_file[0] = str_start_with (libc_ver_end, "/usr/lib/");
+		is_debug_file[1] = str_start_with (libc_ver_end, "/lib/");
 
-	if (!is_debug_file[0] && !is_debug_file[1]) {
-		custom_libc = r_cons_input ("Is a custom library? (LD_PRELOAD=..) Enter full path glibc: ");
-		snprintf (path, sizeof (path), "%s", custom_libc);
-		free (custom_libc);
-		goto arena;			
-	}
+		if (!is_debug_file[0] && !is_debug_file[1]) {
+			custom_libc = r_cons_input ("Is a custom library? (LD_PRELOAD=..) Enter full path glibc: ");
+			snprintf (path, sizeof (path), "%s", custom_libc);
+			free (custom_libc);
+			goto arena;			
+		}
 
-	if (is_debug_file[0]) {
-		snprintf (path, sizeof (path), "%s", libc_ver_end);
-		goto arena;
-	}
+		if (is_debug_file[0]) {
+			snprintf (path, sizeof (path), "%s", libc_ver_end);
+			goto arena;
+		}
 
-	if (is_debug_file[1] && r_file_is_directory ("/usr/lib/debug") && !r_file_is_directory ("/usr/lib/debug/.build-id")) {
-		snprintf (path, sizeof (path), "%s%s", dir_dbg, libc_ver_end);
-	} 
+		if (is_debug_file[1] && r_file_is_directory ("/usr/lib/debug") && !r_file_is_directory ("/usr/lib/debug/.build-id")) {
+			snprintf (path, sizeof (path), "%s%s", dir_dbg, libc_ver_end);
+		} 
 
-	if (is_debug_file[1] && r_file_is_directory ("/usr/lib/debug/.build-id")) {
-		get_hash_debug_file (libc_ver_end, hash, sizeof (hash));
-		libc_ver_end = hash;
-		snprintf (path, sizeof (path), "%s%s%s", dir_dbg, dir_build_id, libc_ver_end);
-	} 
+		if (is_debug_file[1] && r_file_is_directory ("/usr/lib/debug/.build-id")) {
+			get_hash_debug_file (libc_ver_end, hash, sizeof (hash));
+			libc_ver_end = hash;
+			snprintf (path, sizeof (path), "%s%s%s", dir_dbg, dir_build_id, libc_ver_end);
```

Comment:
path = r_str_newf ()... instead of snrpintf . better not to impose path size limits. use sys/limits.h or the heap.